### ISSUE TITLE
ComboBox Properties

### ISSUE THREAD

**Issue by srcarter3** [vincenzopalazzo/material-ui-swing#62]
This is an enhancement request and not an urgent thing by any means.

Our UX team has designed a new look for a combobox:
<img width="209" alt="dropdown" src="https://user-images.githubusercontent.com/21114460/60553020-38bf8f00-9ce6-11e9-8e7a-664ff262a39b.png">

I've been able to get a bit closer to this design by changing the background and foreground colors:
`		UIManager.put("ComboBox.background", theme.getBackgroundPrimary());
		UIManager.put("ComboBox.foreground", theme.getTextColor());
		UIManager.put("ComboBox.selectionBackground", theme.getBackgroundPrimary());
		UIManager.put("ComboBox.selectionForeground", theme.getTextColor());
		UIManager.put("ComboBox.disabledBackground", theme.getHighlightBackgroundPrimary());
		UIManager.put("ComboBox.disabledForeground", theme.getDisabledLabelTextColor());`
![combobox1](https://user-images.githubusercontent.com/21114460/60553062-76241c80-9ce6-11e9-9714-7fc8dc7b0cc7.png)

I tried changing the button background and that worked, but I could not change the button  "border".
`UIManager.put("ComboBox.buttonBackground", theme.getBackgroundPrimary());`
![Screenshot-Add a New Layer-3](https://user-images.githubusercontent.com/21114460/60553100-b1bee680-9ce6-11e9-997c-588a57edf47a.png)

I tried:
`UIManager.put("ComboBox.buttonBorder", BorderFactory.createEmptyBorder());`
But that didn't work.  If that property could be implemented that would be nice.

Also, is it possible to set the icon using the UI Manager?  I think that might be useful as well.  The regular button icon and the disabled icon.

One last thing, would it be possible to pass in a property to draw the border with rounded edges?  Like how JButtons are drawn?  That would be a nice feature.

Anyway, this is just a "nice to have" request.  Thank you!

**Comment by vincenzopalazzo** [comment]
Hi @srcarter3,

I **Love** this effect for the JCoboBox  I want to import this effect for default.

Yes Absolutely yes, we will search a method for creating this effect.

Thanks so much for your help.

Feedback this code have a bug in the JComboBox and another components equals
`UIManager.put("ComboBox.buttonBorder", BorderFactory.createEmptyBorder());`
[This is a issue referiment](https://github.com/atarw/material-ui-swing/issues/74).
*is only information for you*

**Comment by vincenzopalazzo** [comment]
Hi @srcarter3, This is the final effect with the possibility to add the two different color when the component is focused or not focused.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/17150045/60842223-e4c81500-a1d3-11e9-8a8e-7fd7f82a5d41.gif)

for adding this effect you use this constant
```
table.put("ComboBox.unfocusColor", MaterialColors.COSMO_BLACK);
table.put("ComboBox.focusColor", MaterialColors.LIGHT_BLUE_500);
table.put("ComboBox.border", MaterialBorders.roundendLineColorBorder(MaterialColors.COSMO_BLACK));
```

And I have created a new tipe of the border round, you can use it in two mode
- Border rounded Light blue with the constant `MaterialBorders.ROUNDED_CORNER_BORDER`
- A personal color border with the static function `MaterialBorders.roundedLineColorBorder("YOUR_COLOR");`

Unfortunately, the icon change is more complex than expected but it will come

**Comment by srcarter3** [comment]
Okay, awesome. Thanks Vincent. I will check it out and play with it.

**Comment by srcarter3** [comment]
I was able to try out the library with the new combobox. The first thing I noticed, I believe has to do with the round border.  The background of the round border is white, so the text box still looks square.  You need to get the background color of the parent component if it's not null, so that it matches and then you are left with the round border.

![image](https://user-images.githubusercontent.com/21114460/60922900-99088080-a252-11e9-81df-66a57ad13d53.png)

**Comment by vincenzopalazzo** [comment]
Released version beta 1.3, it added this style
